app/controllers/TransferAgreementController.scala

```diff
@@ -43,14 +39,29 @@ class TransferAgreementController @Inject()(val controllerComponents: SecurityCo
         .verifying("All records must be open", b => b)
     )(TransferAgreementData.apply)(TransferAgreementData.unapply)
   )
+  private val options: Seq[(String, String)] = Seq("Yes" -> "true", "No" -> "false")
+  implicit val language: Lang = langs.availables.head
+  private val addTransferAgreementClient = graphqlConfiguration.getClient[AddTransferAgreement.Data, AddTransferAgreement.Variables]()
+
+  def transferAgreement(consignmentId: UUID): Action[AnyContent] = secureAction.async { implicit request: Request[AnyContent] =>
+    val consignmentStatusService = new ConsignmentStatusService(graphqlConfiguration)
 
-  def transferAgreement(consignmentId: UUID): Action[AnyContent] = consignmentExists(consignmentId) { implicit request: Request[AnyContent] =>
-    Ok(views.html.transferAgreement(consignmentId, transferAgreementForm, options))
+    consignmentStatusService.consignmentStatus(consignmentId, request.token.bearerAccessToken).map {
+      consignmentStatus =>
+        val transferAgreementStatus: Option[String] = consignmentStatus.flatMap(_.transferAgreement)
+        transferAgreementStatus match {
+          case Some("Completed") => Ok(views.html.transferAgreementAlreadyConfirmed(consignmentId, transferAgreementForm, options, isAlreadyCompleted = true))
+                                      .uncache()
+          case _ =>  Ok(views.html.transferAgreement(consignmentId, transferAgreementForm, options, isAlreadyCompleted = false))
```

Comment:
The `isAlreadyCompleted` parameter looks redundant now the views are separate.

Reply:
Yes, I'm using passing it into the [formGroup](https://github.com/nationalarchives/tdr-transfer-frontend/pull/924/files#diff-51fb0672e4f7baf91c47a6201f1d28829ad17d7ac936969c7d9b4bcf7f7cc55b) view [here](https://github.com/nationalarchives/tdr-transfer-frontend/pull/924/files#diff-14c3b469abbfcd919a4f7bf521a6dafd9fdbe932289f3ca7bdd4ef23c665e7f9R26) and [here](https://github.com/nationalarchives/tdr-transfer-frontend/pull/924/files#diff-91030991d683ebd29198079140e082333656ea95b8c1a801be98331592819096R21)

The formGroup view uses it to determine whether to disable and select the checkboxes and whether to add the `CSRF.formField` to the form

Reply:
Could the two main templates pass in `true` or `false` directly, rather than you having to pass a parameter from the controller?

Reply:
Sure!